core/src/main/java/org/elasticsearch/cluster/metadata/MetaDataMappingService.java

```diff
@@ -217,9 +218,9 @@ public void refreshMapping(final String index, final String indexUUID) {
                     // failures here mean something is broken with our cluster state - fail all tasks by letting exceptions bubble up
                     for (String index : request.indices()) {
                         final IndexMetaData indexMetaData = currentState.metaData().index(index);
-                        if (indexMetaData != null  && indicesService.hasIndex(index) == false) {
+                        if (indexMetaData != null  && indicesService.hasIndex(indexMetaData.getIndex()) == false) {
```

Comment:
this continues our discussion from yesterday (folded away by github). 

Later on in applyRequest we now have:

```
for (String indexName : request.indices()) {
   Index index = metaData.index(indexName).getIndex();
```

but metaData may not have the index in question because we ignore that case here by doing `indexMetaData != null`.